### PR TITLE
Save favorites

### PR DESCRIPTION
**Pull request by JakeGloschat**

### CODE REVIEW

File: NewBrewer/NewBrewer/Networking/FirebaseService.swift [hunk 1]
```diff
@@ -6,3 +6,64 @@
 //
 
 import Foundation
+import FirebaseFirestore
+import FirebaseStorage
+
+//Make it's own file
```
Comment (maxpoff) [inline NewBrewer/NewBrewer/Networking/FirebaseService.swift]: Are you going to make this its own file?

File: NewBrewer/NewBrewer/Networking/FirebaseService.swift [hunk 2]
```diff
@@ -6,3 +6,64 @@
...
+    func saveBeer(beer: BeerToSave, completion: @escaping (Result<Bool, FirebaseError>) -> Void)
+    func delete(beer: BeerToSave, completion: @escaping (Result<Bool, FirebaseError>) -> Void)
+    func loadBeers(completion: @escaping (Result<[BeerToSave], FirebaseError>) -> Void)
+}
+
+struct FirebaseService: FirebaseServicable {
+    
+    // MARK: - Properties
+    let ref = Firestore.firestore()
+   // let storage = Storage.storage().reference()
```
Comment (maxpoff) [inline NewBrewer/NewBrewer/Networking/FirebaseService.swift]: Remove unnecessary code

File: NewBrewer/NewBrewer/Views/BeerIngredients/BeerIngredientsView.swift [hunk 3]
```diff
@@ -13,15 +13,14 @@ class BeerIngredientsViewController: UIViewController {
...
-        configureAddFavoriteButton()
     }
     
     // MARK: - Properties
     private let tableView = UITableView()
     private let tableViewCellReuseIdentifier = "TableViewCell"
     private let collectionViewCellReuseIdentifier = "CollectionViewCell"
     var beer: Beer?
-    var viewModel: SearchBeerViewModel!
+//    var viewModel: BeerIngredienstViewModel!
```
Comment (maxpoff) [inline NewBrewer/NewBrewer/Views/BeerIngredients/BeerIngredientsView.swift]: remove unnecessary code

File: NewBrewer/NewBrewer/Views/SearchBeer/Custom Cell/BeerListTableViewCellModel.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    private var service: FirebaseServicable
+//    var beers: [BeerToSave]
+    let ref = Firestore.firestore()
+    
+    init(delegate: BeerListTableViewCellModelDelegate, firebaseService: FirebaseServicable = FirebaseService()) {
+        self.delegate = delegate
+        self.service = firebaseService
+    }
+    
+//    func checkSavedBeers() {
```
Comment (maxpoff) [inline NewBrewer/NewBrewer/Views/SearchBeer/Custom Cell/BeerListTableViewCellModel.swift:28]: remove unnecessary code

File: NewBrewer/NewBrewer/Views/SearchBeer/SearchBeerViewModel.swift [hunk 5]
```diff
@@ -36,5 +41,49 @@ class SearchBeerViewModel { // This is a concrete type
...
+    
+    func saveFavoriteBeer(beerToSave: BeerToSave) {
+//        guard favoritedBeers.contains(beerToSave) else {
+        if let indexOfBeer = favoritedBeers.firstIndex(of: beerToSave) {
+            
+            self.favoritedBeers.remove(at: indexOfBeer)
+            deleteSavedBeer(beer: beerToSave)
+            return
+        }
+//            return
```
Comment (maxpoff) [inline NewBrewer/NewBrewer/Views/SearchBeer/SearchBeerViewModel.swift]: remove unnecessary code

**Review by maxpoff — APPROVED** [review]
fix these couple things and you're good to go

File: NewBrewer/NewBrewer/Networking/FirebaseService.swift (hunk 1 above)
Comment (JakeGloschat) [reply NewBrewer/NewBrewer/Networking/FirebaseService.swift]: check

File: NewBrewer/NewBrewer/Networking/FirebaseService.swift (hunk 2 above)
Comment (JakeGloschat) [reply NewBrewer/NewBrewer/Networking/FirebaseService.swift]: check

File: NewBrewer/NewBrewer/Views/BeerIngredients/BeerIngredientsView.swift (hunk 3 above)
Comment (JakeGloschat) [reply NewBrewer/NewBrewer/Views/BeerIngredients/BeerIngredientsView.swift]: Removed the code and removed the view model file.  Changed where the logic was happening so the file is no longer needed.  Thank you.

File: NewBrewer/NewBrewer/Views/SearchBeer/Custom Cell/BeerListTableViewCellModel.swift (hunk 4 above)
Comment (JakeGloschat) [reply NewBrewer/NewBrewer/Views/SearchBeer/Custom Cell/BeerListTableViewCellModel.swift:28]: Again, refactored code to remove logic from the cell to the SearchBeerViewModel so I wasn't doing a call for each cell.  Removed entire TableViewCellModel file.

File: NewBrewer/NewBrewer/Views/SearchBeer/SearchBeerViewModel.swift (hunk 5 above)
Comment (JakeGloschat) [reply NewBrewer/NewBrewer/Views/SearchBeer/SearchBeerViewModel.swift]: check

**Pull request merged**